web/onos-gui/src/app/onos-topo/topodevice.service.ts

```diff
@@ -314,6 +339,23 @@ export class TopoDeviceService {
         return false;
     }
 
+    removeRelation(name: string): boolean {
+        if (this.relationshipsList.has(name)) {
+            this.relationshipsList.delete(name);
+            return true;
+        }
+        return false;
+    }
+
+    addTopoRelation(obj: Object): boolean {
+        const name = obj.getRelation().getSrcEntityId() + ' ' + obj.getRelation().getTgtEntityId();
```

Comment:
the relationship itself has an `id` that you should use as the key in the map - i'm not sure it's safe to assume there will not be 2 relationships with the same `src` and `tgt`